### PR TITLE
 Bad channel from FED error

### PR DESCRIPTION
**Pull request by suchandradutta**
introduced a tool to create SiStrip BAD channels from FED errors in the form of a histogram. Will be used as a  standalone application by Tracker Shifters (not used in any official workflow)

### CODE REVIEW

File: CalibTracker/SiStripESProducers/test/mergeBadChannel_cfg.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+process = cms.Process("BadChannelMerge")
+process.MessageLogger = cms.Service("MessageLogger",
+    cout = cms.untracked.PSet(
+        threshold = cms.untracked.string('INFO')
+    ),
+    destinations = cms.untracked.vstring('cout')
+)
+process.load('Configuration.Geometry.GeometryIdeal_cff')
+process.load("Configuration.Geometry.GeometryReco_cff")
+
```
Comment (mmusich) [inline CalibTracker/SiStripESProducers/test/mergeBadChannel_cfg.py]: @suchandradutta in the spirit of cleaning configuration from legacy modules, could you switch here to `process.load("Configuration.Geometry.GeometryRecoDB_cff")` ?

File: CalibTracker/SiStripESProducers/test/mergeBadChannel_cfg.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+process.load('Configuration.StandardSequences.FrontierConditions_GlobalTag_condDBv2_cff')
+
```
Comment (mmusich) [inline CalibTracker/SiStripESProducers/test/mergeBadChannel_cfg.py]: `condDBv2` is not needed anymore

File: CalibTracker/SiStripESProducers/test/mergeBadChannel_cfg.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+from Configuration.AlCa.GlobalTag_condDBv2 import GlobalTag
```
Comment (mmusich) [inline CalibTracker/SiStripESProducers/test/mergeBadChannel_cfg.py]: here too

File: CalibTracker/SiStripESProducers/test/mergeBadChannel_cfg.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+process.GlobalTag = GlobalTag(process.GlobalTag, 'auto:com10', '')
```
Comment (mmusich) [inline CalibTracker/SiStripESProducers/test/mergeBadChannel_cfg.py]: `auto:run2_data` is more updated

File: CalibTracker/SiStripESProducers/test/fedBadChannelFromNoiseRun_cfg.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        threshold = cms.untracked.string('INFO')
+    ),
+    destinations = cms.untracked.vstring('cout')
+)
+process.load('Configuration.Geometry.GeometryIdeal_cff')
+process.load("Configuration.Geometry.GeometryReco_cff")
+
+process.load('Configuration.StandardSequences.FrontierConditions_GlobalTag_condDBv2_cff')
+
+from Configuration.AlCa.GlobalTag_condDBv2 import GlobalTag
```
Comment (mmusich) [inline CalibTracker/SiStripESProducers/test/fedBadChannelFromNoiseRun_cfg.py]: same comments as below

**Comment by mmusich** [comment]
@suchandradutta is there a 81X equivalent of this PR?

**Comment by mmusich** [comment]
please test

**Comment by suchandradutta** [comment]
preparing a tag for 81X too @mmusich

**Comment by mmusich** [comment]
please test

File: CalibTracker/SiStripESProducers/plugins/services/modules.cc [hunk 6]
```diff
@@ -40,3 +40,7 @@ DEFINE_FWK_SERVICE(SiStripBaseDelayGenerator);
 #include "FWCore/ServiceRegistry/interface/ServiceMaker.h"
 #include "CalibTracker/SiStripESProducers/interface/SiStripConfObjectGenerator.h"
 DEFINE_FWK_SERVICE(SiStripConfObjectGenerator);
+
+#include "FWCore/ServiceRegistry/interface/ServiceMaker.h"
+#include "CalibTracker/SiStripESProducers/interface/SiStripBadModuleFedErrService.h"
+DEFINE_FWK_SERVICE(SiStripBadModuleFedErrService);
```
Comment (mmusich) [inline CalibTracker/SiStripESProducers/plugins/services/modules.cc]: @suchandradutta can you please address David's comment to PR #13664  (https://github.com/cms-sw/cmssw/pull/13664#discussion_r58290500)

**Comment by davidlange6** [comment]
moving the question of @mmusich here - as I had missed it in the closed PR

@davidlange6 can you clarify what is the correct approach? 
I think @suchandradutta reproduced the pattern of other dependent records in the same package where other
dependent ESSources are made (SiStripNoisesFakeESSource, SiStripBadModuleConfigurableFakeESSource,SiStripLorentzAngleFakeESSource etc)
using services like (SiStripNoisesGenerator, SiStripBadModuleGenerator, SiStripLorentzAngleGenerator).
thanks,

Marco


The problem is that all of these services violate the design of services (they affect physics, use esproducts) and thus are subject to being broken at any time by other developments. The ones used in production workflows were flushed out over time and I presume (but it predates my role that started 2.5 years ago) that these services were also indicated at the time as needing work. 

Given that I would not like to add more services into the same category.

**Comment by suchandradutta** [comment]
Can you kindly accept this PR. We had a discussion together (@boudoul , @venturia @davidlange6) and it was agreed that this will be accepted and Tracker will re-organize the package to avoid such usage of service. I do not have anything further to add in this PR.

**Comment by mmusich** [comment]
+1
conditional to https://github.com/cms-sw/cmssw/pull/13888#issuecomment-227548681

**Comment by davidlange6** [comment]
+1

**Pull request merged**